lib/templateFactory.js

```diff
@@ -321,6 +341,69 @@ class TemplateFactory extends BaseFactory {
         return super.get(config);
     }
 
+    /**
+     * Get metrics for the pipelines that use the given template
+     * @method getPipelineUsage
+     * @param  {String}     fullTemplateName    Name of the template and the version or tag (e.g. chef/publish@1.2.3)
+     * @return {Promise}                        List of metrics for the pipelines or null if template is not found.
+     */
+    async getPipelineUsage(fullTemplateName) {
+        const template = await this.getTemplate(fullTemplateName);
+
+        if (!template) {
+            return null;
```

Comment:
Also add a test to cover this scenario
```suggestion
            throw new Error('Template does not exist');
```

Reply:
If the template doesn't exist, shouldn't the UI direct to 404, like it does for getTemplate? With this it will return an internal server error. Or would we catch it in `screwdriver`?

Reply:
A function should consistently return one type of result for all the scenarios or throw error.
Here, the result should always be list of pipelines. If we return `null`, there could ambiguity on what it represent: no pipelines using the template vs no such template.

Yes, catch it in `screwdriver` and `throw boom.notFound(error)` when the error is `Template does not exist`.